plugin/core/url.py

```diff
@@ -47,6 +47,21 @@ def uri_to_filename(uri: str) -> str:
         return url2pathname(parsed.path)
 
 
+def parse_uri(uri: str) -> Tuple[bool, str]:
+    """
+    Parses an URI into a tuple where the first element is True if the URI is a
+    file URI, otherwise False. The second element is the local filesystem path
+    if the URI is a file URI, otherwise the second element is the original URI.
+    """
+    parsed = urlparse(uri)
+    if parsed.scheme == "file":
+        if os.name == 'nt':
+            # TODO: this is wrong for UNC paths
+            return True, url2pathname(parsed.path).strip('\\')
+        return True, url2pathname(parsed.path)
+    return False, uri
```

Comment:
Well, this doesn't return the schema. Many places (in LSP at least) would benefit from a helper function like that if it would return the schema. I'm not sure if external code (helper packages) would benefit from knowing the schema. If not then maybe it's OK like that...

Can we also export this function in public plugin API?

Reply:
> Can we also export this function in public plugin API?

Yes, I think this should be the public facing function for handling URIs.

Reply:
I'm still not totally happy with this function. If it's not a file URI, then you might need the URI fragment instead (for example). But then you would have to use the low-level urllib.parse functions again.

Reply:
It probably would be OK to provide two separate functions. Anything is better than using urlparse directly :)